src/prism.c

```diff
@@ -2172,11 +2172,16 @@ pm_call_target_node_create(pm_parser_t *parser, pm_call_node_t *target) {
 static pm_index_target_node_t *
 pm_index_target_node_create(pm_parser_t *parser, pm_call_node_t *target) {
     pm_index_target_node_t *node = PM_ALLOC_NODE(parser, pm_index_target_node_t);
+    pm_node_flags_t flags = target->base.flags;
+
+    if (parser->version != PM_OPTIONS_VERSION_CRUBY_3_3_0) {
+        flags |= PM_CALL_NODE_FLAGS_ATTRIBUTE_WRITE;
+    }
```

Comment:
@andrykonchin and I are looking at which version we'd set.
We do want the flag to be set here.
But with the current that means we need to use PM_OPTIONS_VERSION_LATEST.

OTOH we do not want `it` handling from 3.4 (https://github.com/ruby/prism/pull/2124).
And using `PM_OPTIONS_VERSION_LATEST` would enable `it`.

So how to solve this?
Maybe we need `PM_OPTIONS_VERSION_CRUBY_3_3` (without _0) to mean 3.3 syntax with the latest fixes?
Alternatively, is it worth hiding this flag for Prism versions newer than the one shipped with 3.3.0? Maybe not?
It feels like this flag should not hurt anything, and if someone wants exactly the version of Prism as shipped in 3.3.0 then they can just specify that Prism release in their Gemfile.

Reply:
@kddnewton WDYT?
Feel free to extract this to an issue/discussion if you prefer to discuss it somewhere else.

Reply:
Maybe the fundamental question is whether this change would go in CRuby 3.3.1. I think it should as a bug fix.